core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1581,6 +1646,171 @@ private void validateTableModelAndCreateTypeCast(
             throw SqlException.position(0).put("timestamp is not defined");
         }
     }
+    
+    private CompiledQuery sqlBackup(SqlExecutionContext executionContext) throws SqlException {
+        if (null == configuration.getBackupRoot()) {
+            throw CairoException.instance(0).put("Backup is disabled, no backup root directory is configured in the server configuration ['cairo.sql.backup.root' property]");
+        }
+
+        final CharSequence tok = SqlUtil.fetchNext(lexer);
+        if (null != tok) {
+            if (Chars.equalsLowerCaseAscii(tok, "table")) {
+                return sqlTableBackup(executionContext);
+            }
+            if (Chars.equalsLowerCaseAscii(tok, "database")) {
+                return sqlDatabaseBackup(executionContext);
+            }
+        }
+
+        throw SqlException.position(lexer.getPosition()).put(" expected 'table' or 'database'");
+    }
+
+    private CompiledQuery sqlDatabaseBackup(SqlExecutionContext executionContext) throws SqlException {
+        setupBackupRenamePath();
+        // TODO: Cache the nativeLPSZ
+        NativeLPSZ nativeLPSZ = new NativeLPSZ();
+        ff.iterateDir(path.of(configuration.getRoot()).$(), (file, type) -> {
+            nativeLPSZ.of(file);
```

Comment:
Lambda should be refactored into a method, will play nicely with caching nativeLPSZ